### PR TITLE
[feat]PIP 167: Make it Configurable to Require Subscription Permission for Consumer

### PR DESCRIPTION
**Pull request by michaeljmarshall**
This PR is meant to fulfill the requirements described by https://github.com/apache/pulsar/issues/15597.

The implementation essentially locks down all subscriptions so that in order to consume from a topic, a consumer will need permission granted for the specific topic and for the subscription within the namespace. A tenant admin and a super user will not need additional permission.

### CODE REVIEW

File: pulsar-websocket/src/main/java/org/apache/pulsar/websocket/service/WebSocketProxyConfiguration.java [hunk 1]
```diff
@@ -120,6 +120,15 @@ public class WebSocketProxyConfiguration implements PulsarConfiguration {
...
     private boolean authorizationAllowWildcardsMatching = false;
 
+    @FieldContext(
+            doc = """
+                    If a namespace has no roles configured in the subscription permission for a given subscription name,
+                    allow all roles that have permission to consume a the topic to consume from the subscription.
+                    See Namespaces#grantPermissionOnSubscription in the Java Admin API Client for details on granting
+                    permission.
+                    """)
+    private boolean allowImplicitSubscriptionAuthenticationPermission = true;
```
Comment (eolivelli) [inline pulsar-websocket/src/main/java/org/apache/pulsar/websocket/service/WebSocketProxyConfiguration.java:130]: This parameter seems not referenced in other places of the patch

Comment (michaeljmarshall) [reply pulsar-websocket/src/main/java/org/apache/pulsar/websocket/service/WebSocketProxyConfiguration.java]: Thanks, I missed that when doing a find and replace. Fixed a new commit.

**Comment by michaeljmarshall** [comment]
@eolivelli @dlg99 @nicoloboschi - this is ready for review

**Review by nicoloboschi — APPROVED** [review]
LGTM

**Review by dlg99 — APPROVED** [review]
LGTM

**Comment by michaeljmarshall** [comment]
CI was failing due to invalid caching. I merged `datastax/3.1_ds` to try to trigger cache invalidation.

**Pull request merged**